### ISSUE TITLE
Best Traineddata Feedback - Devanagari Script

### ISSUE THREAD

**Issue by Shreeshrii** [tesseract-ocr/tessdata#69]
Devanagari Script Best Traineddata  has the complete English alphabet. the wordlist also has a complete English wordlist including words in ALL CAPS.

@theraysmith What is the logic for this? It will be useful to have 0-9 numbers in Latin script as part of Devanagari traineddata, but why full English support? I think this reduces accuracy...

**Comment by theraysmith** [comment]
For Latin, I have ~4500 fonts to train with.
For Devanagari ~50, and for Kannada 15.
With a theory that poor accuracy on test data and over-fitting on training data was caused by the lack of fonts, I tried mixing the training data with English, thinking that English is often mixed in anyway, and some of the font diversity might generalize to the other script. The overall effect was slightly positive, so I left it that way.
It didn't resolve the issue with poor accuracy on test data, and even though a lot of languages are affected, I think there are multiple problems at work: with Hebrew/Arabic it's points, and with (at least some of) the Indic scripts, it looks like it might be older styles of typography.

**Comment by Shreeshrii** [comment]
I will list fonts with older orthography later.

Hope you have already got the ones listed at https://github.com/tesseract-ocr/tesseract/wiki/Fonts

There are also a large number of fonts available from http://ildc.in/Hindi/GIST/hindi_cd_2/windows/index.htm
It is part of a larger zipped file.

Is there a method for creating box/tiff from scanned images? We can use it create more variations for training.

**Comment by Shreeshrii** [comment]
Fonts with older/northern orthography

Santipur OT
Siddhanta-Calcutta
Sahadeva
Uttara

Some older orthography alphabet images are at
https://github.com/Shreeshrii/tess4eval_deva/tree/master/images
with corresponding ground truth at
https://github.com/Shreeshrii/tess4eval_deva/tree/master/gt

Also, more variations of fonts can be created via exposures -3, -2, -1, 0, 1 and by the additional program to add more noise to the images (you had mentioned adding the feature to text2image).

------
If I want to experiment further for Devanagari training - 

* how many lines of text do you suggest to use for finetuning? 
* how may iterations? Does it need to go to 0.01%

* Should the training text be same for all fonts or should I use different text but same number of lines for each?

* Do the wordlists get generated from the training data? Can the word dawgs be replaced by u

**Comment by Shreeshrii** [comment]
@theraysmith 

Just saw https://github.com/tesseract-ocr/tesseract/blob/2633fef0b6ac9b616eae3d457bf796076eb8f43c/training/tesstrain_utils.sh#L215

> common_args+=" --outputbase=${outbase} --max_pages=3"

Does that mean that recommended training_text size for finetuning, replace a layer is about 150-200 lines?

**Comment by Shreeshrii** [comment]
For an example image with a mix of Hindi and Sanskrit text and a lot of punctuation - quotes, hyphens etc, I found that Devanagari traineddata gave best result. Though there  are a few conjunct consonants that are NOT being recognized.

eg.
```
ह्र
ह्न
घ्र
```

Image and ground truth are attached.

![vallabhamahaganapatitrishatinamavali0](https://user-images.githubusercontent.com/5095331/29660677-86ad651a-88df-11e7-9f85-a09c078debb2.png)
[vallabhamahAgaNapatitrishatinAmAvalI0-gt.txt](https://github.com/tesseract-ocr/tessdata/files/1248378/vallabhamahAgaNapatitrishatinAmAvalI0-gt.txt)

**Comment by Shreeshrii** [comment]
Some more testing of Devanagari script traineddata vs san and hin.

While in most cases, Devanagari does better. In samples with large font size 48 px, accuracy drops to 50% or lower.

See attached reports, run using https://github.com/eddieantonio/isri-ocr-evaluation-tools which supports utf-8 text.

[ALL-accuracy-imageshin.txt](https://github.com/tesseract-ocr/tessdata/files/1274356/ALL-accuracy-imageshin.txt)


[ALL-deva-imageshin-rpt.txt](https://github.com/tesseract-ocr/tessdata/files/1274355/ALL-deva-imageshin-rpt.txt)

[ALL-deva-imagessan-rpt.txt](https://github.com/tesseract-ocr/tessdata/files/1274358/ALL-deva-imagessan-rpt.txt)

**Comment by Shreeshrii** [comment]
best/mar does better than best/Devanagari for Marathi text.

[ALL-deva-imagesmar-rpt.txt](https://github.com/tesseract-ocr/tessdata/files/1274520/ALL-deva-imagesmar-rpt.txt)

**Comment by christophered** [comment]
@Shreeshrii `and by the additional program to add more noise to the images`
How to add more noise to the generated images?

**Comment by stweil** [comment]
@jbarth-ubhd has trained a Devanagari model which gives better results in my first test. Maybe he can share it on [tessdata_contrib](https://github.com/tesseract-ocr/tessdata_contrib).

**Comment by Shreeshrii** [comment]
@stweil I am curious to know how much improvement was achieved via finetuning. 

See related posts in forum - https://groups.google.com/forum/#!msg/tesseract-ocr/NNZ7GOBLB_8/IMqn2IgzAwAJ
and
https://www.iias.asia/the-newsletter/article/naval-kishore-press-digital-hidden-treasure-open-access

**Comment by stweil** [comment]
I only tested a single page where tessdata_best/script/Devanagari achieved 93 % character recognition rate while Jochen's model recognized 98 %.

**Comment by Shreeshrii** [comment]
98% accuracy is very impressive!!

**Comment by Shreeshrii** [comment]
>I only tested a single page where tessdata_best/script/Devanagari achieved 93 % character recognition rate while Jochen's model recognized 98 %.

@stweil, Did you run any additional tests on Jochen's model?

**Comment by stweil** [comment]
No, I didn't.

**Comment by lokeshh** [comment]
@Shreeshrii @stweil Where can I find Jochen's model with 98% accuracy?
